fda/boxplot.py

```diff
@@ -0,0 +1,585 @@
+"""Functional Data Boxplot Module.
+
+This module contains the functions to construct the functional data boxplot.
+It also defines a class for representing the information returned by those functions.
+
+"""
+import matplotlib
+import matplotlib.pyplot as plt
+import math
+
+from .depth_measures import *
+from .grid import FDataGrid
+
+__author__ = "Amanda Hernando Bernabé"
+__email__ = "amanda.hernando@estudiante.uam.es"
+
+def _generic_boxplot_checks(fdatagrid, fig, ax, method, nrows, ncols, colormap, factor):
+
+    fig, ax = fdatagrid.generic_plotting_checks(fig, ax, nrows, ncols)
+
+    if ((method != fraiman_muniz_depth) & (method != band_depth) & (method != modified_band_depth)):
+        raise ValueError("Undefined function.")
+
+    if not isinstance(colormap, matplotlib.colors.LinearSegmentedColormap):
+        raise ValueError("colormap must be of type matplotlib.colors.LinearSegmentedColormap")
+
+    if factor < 0:
+        raise ValueError("The number used to calculate the outlying envelope must be positive.")
+
+    return fig, ax
+
+
+def fdboxplot(fdatagrid, fig=None, ax = None, method=modified_band_depth, prob=[0.5], fullout=False, factor=1.5,
+              colormap=plt.cm.get_cmap('RdPu'), barcol="blue", outliercol="red", mediancol="black", nrows=None, ncols=None):
+    """Implementation of the functional boxplot.
+
+    It is an informative exploratory tool for visualizing functional data, as well as
+    its generalization, the enhanced functional boxplot. Only supports 1 dimensional
+    domain functional data.
+
+    Based on the center outward ordering induced by a :ref:`depth measure <depth-measures>`
+    for functional data, the descriptive statistics of a functional boxplot are: the
+    envelope of the 50% central region, the median curve,and the maximum non-outlying envelope.
+    In addition, outliers can be detected in a functional boxplot by the 1.5 times the 50%
+    central region empirical rule, analogous to the rule for classical boxplots.
+
+    Args:
+        fdatagrid (FDataGrid): Object to be visualized.
+        fig (figure object, optional): figure over with the graphs are plotted in case ax is not specified.
+            If None and ax is also None, the figure is initialized.
+        ax (list of axis objects, optional): axis over where the graphs are plotted. If None, see param fig.
+        method (:ref:`depth measure <depth-measures>`, optional): Method used to order the data.
+            Defaults to :func:`modified band depth <fda.depth_measures.modified_band_depth>`.
+        prob (list of float, optional): List with float numbers (in the range from 1 to 0) that indicate which central regions to
+            represent. Defaults to [0.5] which represents the 50% central region.
+        fullout (boolean): If true, the entire curve of the outlier samples is shown. Defaulsts to False.
```

Comment:
It is the opposite, actually. Also, the name is badly chosen. My first impression was that this parameter required the full curve to be outside the outlying envelope, instead of only a point, to be considered an outlier.

Reply:
A sample is considered an outlier if any of its points falls outside the outlying envelope.
And I know that fullout should be the other way, but I took this parameter from the method implemented in R which has the following description:
fullout: logical for plotting outlying curves.  If FALSE (the default) then only the part
outside the box is plotted. If TRUE, complete outling curves are plotted